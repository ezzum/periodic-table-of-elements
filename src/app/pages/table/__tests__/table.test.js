import configureStore from "redux-mock-store";
import { mount } from "enzyme";
import { Provider } from "react-redux";
import React from "react";

import Table from "../table";

jest.mock("../__elements/elements-grid", () => "div");
jest.mock("../__column-numbers/column-numbers", () => "div");
jest.mock("../__row-numbers/row-numbers", () => "div");
jest.mock("../../../spinner/spinner", () => "div");

describe("Table tests", () => {
  const initialStore = {
    tableProps: {
      columns: 18,
      rows: 7,
    },
    loading: true,
  };
  const mockStore = configureStore();
  let store, wrapper;

  beforeEach(() => {
    store = mockStore(initialStore);
    wrapper = mount(
      <Provider store={store}>
        <Table />
      </Provider>
    );
  });

  it("snapshot test", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
