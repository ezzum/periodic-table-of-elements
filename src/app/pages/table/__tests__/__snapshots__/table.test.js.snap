// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table tests snapshot test 1`] = `ReactWrapper {}`;
